internal/authz/kessel/kessel.go

```diff
@@ -69,3 +72,41 @@ func (a *KesselAuthz) DeleteTuples(ctx context.Context, r *kessel.DeleteTuplesRe
 	}
 	return a.TupleService.DeleteTuples(ctx, r, opts...)
 }
+
+func (a *KesselAuthz) SetWorkspace(ctx context.Context, id int64, workspace, namespace, name string) (*kessel.CreateTuplesResponse, error) {
+	if workspace == "" {
+		return nil, fmt.Errorf("workspace is required")
+	}
+
+	// TODO: remove previous tuple for workspace
```

Comment:
Yes this needs to be done or else a resource would be in multiple workspaces which is a constraint violation.

However, this needs to be done in one atomic call which the relations API doesn't support yet. I'll file a ticket for resolving this and the relations API additions necessary.

And as of now as we know, there is no workspace movement needed, so omitting this should suffice for the next few weeks or so.